Surveys

romanbond opened the pull request:
Changes on our branch involve the new route for the surveys page, survey result menu item and the survey view folder with our views and python files needed to pull from google sheets.

blackjack26 commented:
could you please rebase onto the current version of develop just so you have the most recent changes.

blackjack26 commented on views/surveys/view-survey-results.php:
```diff
@@ -0,0 +1,238 @@
+<?php
```
Rename this file to `view_survey_results.php`

blackjack26 commented on views/surveys/view-survey-results.php:
```diff
@@ -0,0 +1,238 @@
+			<div class="col-sm col-lg-6">
+		<form method="POST" action="/surveys/results" onsubmit = "return validateForm()"name = "userInput">
+		<div class="form-group">
+		<select class="form-control" id="classes" name="classes">
```
Populate with actual classes using PHP not temp "fake" values

blackjack26 commented on views/surveys/view-survey-results.php:
```diff
@@ -0,0 +1,238 @@
+		<div class="form-group">
+		<select class="form-control" name="Year" id="year">
+		<option selected disabled="disabled" value="">Year</option>
+			<option value="2017">2017</option>
```
Make this non-static. When 2018 happens we'll have to update this. Look at the reports team's code. They have a drop down with years in it.

blackjack26 reviewed (changes requested):
I rebased for you but there are some things that can be updated.

evanmena commented on views/surveys/old_scripts/test.py:
```diff
@@ -0,0 +1,10 @@
+#!/user/bin/env python
```
Probably just remove this file.

romanbond commented:
In my last commit I changed the view survey results file name, made the year for the date dynamic and added real classes to the drop down, they're static but I'm working on pulling them with php.

evanmena commented on public/menu.php:
```diff
@@ -24,18 +24,58 @@ function isActive($urls) {
 ?>
 <div class="side-menu">
     <nav class="nav flex-column flex-nowrap">
+<<<<<<< HEAD
```
Clean up the `git merge` lines, ln 73 and 78 as well.

romanbond commented:
Everything should be good to go, the new page now pulls the classes straight from the DB and the date is dynamic.

blackjack26 commented on routes.php:
```diff
@@ -53,3 +55,10 @@
+$router->add('/record-attendence', '/record-attendence/record-attendence.php', 'Record Attendence');
+
+#Survey Page
+$router->add('/surveys','surveys/view-survey-results.php', 'Survey Results');
```
This file was renamed to `view_survey_results.php`, make sure to change that here too.

blackjack26 commented on views/surveys/results.php:
```diff
@@ -0,0 +1,302 @@
+$db->prepare("get_other_classes",
+    "SELECT * FROM classes WHERE topicname NOT IN (" .
+    "SELECT topicname FROM curriculumclasses WHERE curriculumid = $1" .
+    ") ORDER BY topicname");
```
This is an old query, database does not use the `topicname` as the identifier for a class anymore. Make sure to use the most up to date database that uses `classid`.

blackjack26 commented on views/surveys/results.php:
```diff
@@ -0,0 +1,302 @@
+	
+		$db_connection = pg_connect("host=10.11.12.24 dbname=Actual user=postgres password=password");
+		$query = "SELECT participantname FROM surveys WHERE cast(starttime as text) LIKE '" . $_POST["Year"] . "-" . $_POST["Month"] . "-" . $_POST["Day"] . "%' and topicname = '" . $_POST["classes"] . "' ORDER BY participantname;";
+		$result = pg_query($db_connection,$query);
```
You do not need to create a database connection, this has already been done for you. All you need to do is simply do:
```php
$db->query("SELECT ...", [$param1, $param2]);
```
See the other files for an example.

blackjack26 commented on views/surveys/results.php:
```diff
@@ -0,0 +1,302 @@
+												</div>
+												'); 
+				$query2 = "SELECT * FROM surveys WHERE participantname='$col_value' and cast(starttime as text) LIKE '" . $_POST["Year"] . "-" . $_POST["Month"] . "-" . $_POST["Day"] . "%' and topicname = '" . $_POST["classes"] . "';";
+				$result2 = pg_query($db_connection,$query2);
```
Use the pre-existing database connection `$db`.

blackjack26 commented on views/surveys/results.php:
```diff
@@ -0,0 +1,302 @@
+			<?php
+		}
+		pg_free_result($result);
+		pg_close($db_connection);
```
No need for `pg_close()` when using the pre-defined `$db` variable.

blackjack26 commented on views/surveys/view_survey_results.php:
```diff
@@ -0,0 +1,259 @@
+}
+# Select page to display
+if (!empty($params) && $params[0] == 'view') {
+	$view->display('/view-survey-results.php');
```
Not sure why this if-statement exists since this file is `view_survey_results.php`. Unless the route is `/surveys/view`. But even then this page doesn't exist anymore since it was renamed to the current file.

blackjack26 commented on views/surveys/view_survey_results.php:
```diff
@@ -0,0 +1,259 @@
+<?php
+global $params, $route, $view;
+include ('../models/Notification.php');
+$pages = ['view','edit','create','delete','restore'];
```
Change these values to reflect the pages you can actually go to. In your case it looks to only be `'results'`. If you are not using routing parameters at all, there is no need to have lines 4-15.

blackjack26 commented on views/surveys/view_survey_results.php:
```diff
@@ -0,0 +1,259 @@
+	global $db;
+	$filter = "";
+	
+	if ($_SERVER['REQUEST_METHOD'] == 'POST') {
```
Are you ever using a filter on this page? This purpose of this if statement is the filter you see on the curriculum page at the top. You don't need this whole if-statement.

blackjack26 commented on views/surveys/view_survey_results.php:
```diff
@@ -0,0 +1,259 @@
+		<?php
+			function updateDB(){	
+				//$command = escapeshellcmd('py C:\Program Files (x86)\EasyPHP-Devserver-17\eds-www\views\surveys\pullAndParseSurveys.py');
+				$output = shell_exec('py "C:\Program Files (x86)\EasyPHP-Devserver-17\eds-www\views\surveys\pullAndParseSurveys.py"');
```
Probably shouldn't use an absolute path since this is not how it will be configured on the server. If you need to do an absolute path, add configuration to the `config.ini` file that specifies this file path.

blackjack26 commented on views/surveys/view_survey_results.php:
```diff
@@ -0,0 +1,259 @@
+		
+		<!--<form method="POST">
+		<input name="updated" value="Update Surveys" type="Submit" class="btn btn-success" style="float: right;"/>
+		</form>-->
```
Remove this commented out `<form>`

blackjack26 commented on views/surveys/view_survey_results.php:
```diff
@@ -0,0 +1,259 @@
+			<?php
+			$db_connection = pg_connect("host=10.11.12.24 dbname=Actual user=postgres password=password");
+			$query = "SELECT topicname FROM classes";
+			$result = pg_query($db_connection,$query);
```
Use the `$db` connection variable instead of creating your own connection.

blackjack26 commented on views/surveys/view_survey_results.php:
```diff
@@ -0,0 +1,259 @@
+				
+				
+				}
+			}
```
From lines 102-112. It should really look like:

```php
$results = $db->query('SELECT classid, topicname FROM classes WHERE df IS FALSE', []);
if ($results && pg_result_error_field($result, PGSQL_DIAG_SQLSTATE) == 0) {
    while ($class = pg_fetch_assoc($results)) {
        echo '<option value="' . $class['classid'] . '">' . $class['topicname'] . '</option>';
    }
}

blackjack26 commented on views/surveys/view_survey_results.php:
```diff
@@ -0,0 +1,259 @@
+			<option value="2002">2002</option>
+			<option value="2001">2001</option>
+			<option value="2000">2000</option>
+		</select>-->
```
Remove commented out code

blackjack26 commented on views/surveys/view_survey_results.php:
```diff
@@ -0,0 +1,259 @@
+					</script>
+					<p>
+					<form method="POST">
+						<input name="updated" value="Update Surveys" type="Submit" class="btn btn-success" style="background: #5C639A;"/>
```
Use button class `cpca` instead:

`class = "btn cpca"`

blackjack26 reviewed (changes requested):
Also, I'm not sure what the `old_scripts` folder is but, if its old stuff, just remove it altogether.

evanmena commented on routes.php:
```diff
@@ -35,6 +35,8 @@
 $router->add('/classes', 'classes/classes.php', 'Classes');
 $router->add('/locations', 'locations/locations.php', 'Locations');
 
+<<<<<<< HEAD
```
More `git merge` noise in this file. ln 38, 39, 59

evanmena commented on views/surveys/README.md:
```diff
@@ -0,0 +1,32 @@
+# Surveys
+
+* Surveys are hosted on Google Drive using a Google Form which then sends the submitted form data to a linked Google Spreadsheet. 
+<<<<<<< HEAD
```
`git merge` artifacts ln 4, 6, 32. Message me if you need help with git to avoid this.

evanmena commented on views/surveys/pullAndParseSurveys.py:
```diff
@@ -0,0 +1,77 @@
+	#print "ROW:  %s   ----    " % row
+	if (x <= y[0][0]):
+		x += 1
+		
```
You shouldn't mix tab/space whitespace in python. Here you have tabs, above is spaces.

evanmena commented on views/surveys/setup.py:
```diff
@@ -0,0 +1,8 @@
+import pip
```
You can use `pip install -r` to do the heavy lifting. https://devcenter.heroku.com/articles/python-pip shows how the syntax of a `requirements.txt` so that you can do `pip install -r requirements.txt`

blackjack26 committed:
Surveys page

- Python files to pull surveys from Google
- PHP web display for surveys
- Replaced raw text connection info with shell_exec arguments from config.ini

blackjack26 reviewed (approved):
I rebased and made some changes to the PHP pages because the some of the HTML had errors in it, mismatching tags, and wasn't indented nicely. The new code is pushed here and off the newest develop branch. There are still some enhancements with error handling but that can come after this is merged.

evanmena commented on views/surveys/results.php:
```diff
@@ -0,0 +1,192 @@
+    function downloadPDFClass(){
+        var originalContents = document.body.innerHTML;
+        $('.collapse').collapse('toggle');    
+        setTimeout(function(){
```
I'm not entirely sure what's going on here, but it seems like for 200 ms you are replacing the contents of the page with the contents you want to print then swapping it back afterwards. There's definitely an easier way. See `window.print()` combined with [@media print](https://github.com/Capping-CPCA/Web-App/blob/d247be695cfbad73873b996e2e8b3fa4adb0e80f/views/reports/half_year.php#L214) css

Hoenn commented on views/surveys/pullAndParseSurveys.py:
```diff
@@ -4,12 +4,13 @@
-address = "10.11.12.24"
-googleEmail = "pokcpcapep@gmail.com"
-googlePassword = "Marist1234"
+# config.ini information is passed in as arguments from php shell_exec
```
This lets us hide things like the raw text gmail login and password, storing them in `config.ini` which is on the `.gitignore` list.

evanmena replied:
Hides the raw text email and password which as sensitive. `config.ini` is on the `.gitignore` list so it won't show up publicly on the repo.
